### PR TITLE
(WIP) - add first iteration of new shopify endpoints

### PR DESCRIPTION
**Pull request by Dimi727**
* Easy implementation for `locations`, `price_rules ` and `marketing_events` (response is empty though)
* inventory_items - not implemented yet but retrieved successfully in the terminal. Needs 2 extra iterations over other endpoints
* inventory_levels - implemented, but does not work. Need to pass somehow `location_ids` to the request. The implemented function alone does not work. Have to probably implement it for the locations endpoint as an extra function like for orders?

### CODE REVIEW

File: ewah/hooks/shopify.py [hunk 1]
```diff
@@ -208,22 +212,52 @@ def get_data(
...
+            # call location ids
+            locations_url = self._BASE_URL.format(
+                shop=shop_id,
+                version=version,
+                object="locations",
+            )
+            req_kwargs = {
+                "headers": headers,
+            }
+            response = requests.get(locations_url, **req_kwargs)
```
Comment (soltanianalytics) [inline ewah/hooks/shopify.py:228]: Suggestion (non-blocking): No need to do the whole `req_kwargs` stuff here :)

```suggestion
            response = requests.get(locations_url, headers=headers)
```

File: ewah/hooks/shopify.py [hunk 2]
```diff
@@ -261,11 +295,22 @@ def get_data(
...
                 url = response.headers["Link"][1:-13]
+            elif ids_list and shopify_object == 'inventory_levels':
+                # after pagination complete we restart the requests while
+                # we still have ids in id_list
+                finished_pagination = True
+                url = self._BASE_URL.format(
+                    shop=shop_id,
+                    version=version,
+                    object="inventory_levels",
+                )
```
Comment (soltanianalytics) [inline ewah/hooks/shopify.py:312]: Comment (non-blocking): This should not be necessary. Can you check?

**Review by soltanianalytics — APPROVED** [review]
Some non-blocking comments, let me know if you want to discuss any. But we can also merge as is for now :)

File: ewah/hooks/shopify.py (hunk 2 above)
Comment (Dimi727) [reply ewah/hooks/shopify.py:312]: Lets discuss

Comment (soltanianalytics) [reply ewah/hooks/shopify.py:312]: As discussed, my comment is invalid and your code is correct as is :)

File: ewah/hooks/shopify.py [hunk 3]
```diff
@@ -208,22 +212,52 @@ def get_data(
...
+            locations_url = self._BASE_URL.format(
+                shop=shop_id,
+                version=version,
+                object="locations",
+            )
+            req_kwargs = {
+                "headers": headers,
+            }
+            response = requests.get(locations_url, **req_kwargs)
+            locations = response.json().get("locations")
```
[the remark this answers is not in the document]
Comment (Dimi727) [reply ewah/hooks/shopify.py]: done :)

**Review by soltanianalytics — APPROVED** [review]

**Pull request merged**